src/main/java/com/eprosima/fastcdr/idl/templates/TypesSourcev1.stg

```diff
@@ -0,0 +1,2469 @@
+           (s <= 32) ? Size::UInt32: Size::UInt64;
+}
+
+template<Size s>
+struct FindTypeH;
+
+template<>
+struct FindTypeH<Size::UInt8> {
+    using type = std::uint8_t;
+};
+
+template<>
+struct FindTypeH<Size::UInt16> {
+    using type = std::uint16_t;
+};
+
+template<>
+struct FindTypeH<Size::UInt32> {
+    using type = std::uint32_t;
+};
+
+template<>
+struct FindTypeH<Size::UInt64> {
+    using type = std::uint64_t;
+};
+}
+
+template<int S>
+struct FindType {
+    using type = typename internal::FindTypeH<internal::get_size(S)>::type;
+};
+}
+
+$if(ctx.anyCdr)$
+$ctx.setCdrv1Templates$
+$ctx.types:{ type | $if(type.typeCode.isStructType)$#define $type.typeCode.cScopedname$_max_cdr_typesize $type.typeCode.maxSerializedSize$ULL;$endif$}; separator="\n"$
+$ctx.unsetCdrv1Templates$
+$endif$
+
+$extensions : { extension | $extension$}; separator="\n"$
+
+$if(ctx.generateTypesC)$
+$ctx.typeCodesToDefine : { type |
+$sequences_definition(type.value)$
+}; separator="\n"$
+$endif$
+
+$definitions; separator="\n"$
+
+$if(ctx.cdr_both)$
+#endif // FASTCDR_VERSION_MAJOR == 1
+$endif$
+>>
+
+module(ctx, parent, module, definition_list) ::= <<
+namespace $module.name$ {
+
+$definition_list$
+
+} // namespace $module.name$
+>>
+
+definition_list(definitions) ::= <<
+$definitions; separator="\n"$
+
+>>
+
+annotation(ctx, annotation) ::= <<>>
+
+interface(ctx, parent, interface, export_list) ::= <<
+$export_list$
+>>
+
+export_list(exports) ::= <<
+$exports; separator="\n"$
+
+>>
+
+exception(ctx, parent, exception) ::= <<
+$if(ctx.printexception)$
+$exception.name$::$exception.name$()
+    : UserException()
+{
+    $exception.members : { member |$member_default_init(ctx=ctx, member=member, loopvar=ctx.newLoopVarName)$}; separator="\n"$
+}
+
+$exception.name$::$exception.name$(
+        const $exception.name$& x)
+    : UserException(x)
+{
+    $exception.members : { member |$member_copy(ctx=ctx, member=member)$}; separator="\n"$
+}
+
+$exception.name$::$exception.name$(
+        $exception.name$&& x)
+    : UserException(std::move(x))
+{
+    $exception.members : { member |$member_move(member=member)$}; separator="\n"$
+}
+
+$exception.name$& $exception.name$::operator =(
+        const $exception.name$& x)
+{
+    UserException::operator =(x);
+    $exception.members : { member |$member_copy(ctx=ctx, member=member)$}; separator="\n"$
+    return *this;
+}
+
+$exception.name$& $exception.name$::operator =(
+        $exception.name$&& x)
+{
+    UserException::operator =(std::move(x));
+    $exception.members : { member |$member_move(member=member)$}; separator="\n"$
+    return *this;
+}
+
+$exception.name$::~$exception.name$() throw()
+{
+    $exception.members : { member |$varDel(variable=member)$}; separator="\n"$
+}
+
+void $exception.name$::raise() const
+{
+    throw *this;
+}
+
+$if(ctx.anyCdr)$
+size_t $exception.name$::getCdrSerializedSize(
+        const $exception.name$& data,
+        size_t current_alignment)
+{
+    (void)data;
```

Comment:
```suggestion
    static_cast<void>(data);
```

Please, apply to other C-type castings if possible.

Reply:
Done in 020947b